### DISCUSSION TITLE
Would it be possible add uBlock Origin as a userscript or similar?

### DISCUSSION THREAD

**Discussion by ProtoBelisarius** [minbrowser/min#2486] [Q&A]
I really like this browser, but I find myself dearly missing uBlock Origin. Even with Content Blocking set to all ads and trackers, it still leaves out quite a few. 

Is there a way to get it working here or are there fundamental road blocks that would doom this from the start?

**Comment by PalmerAL** [comment]
Userscripts run inside the context of the page, so they wouldn't have the access needed to block network requests.

You could implement cosmetic filtering with a userscript (where nothing is stopped from loading, but the ad is visually hidden) - that's something that uBlock origin supports that we don't.

Do you have some examples of ads that Min doesn't block? It's possible there's something we can improve there.

**Reply by ghost** [reply]
I don't think it blocks anything on Polish sites

**Reply by PalmerAL** [reply]
Can you provide a specific URL where things aren't blocked?

**Reply by ghost** [reply]
https://www.wp.pl/

**Reply by PalmerAL** [reply]
Actually, uBlock Origin doesn't seem to work on that site either, at least with the filters I have enabled:

![Screenshot 2025-04-15 at 10 21 43 PM](https://github.com/user-attachments/assets/11140592-6548-4212-b48e-e8a54088c4e6)

It looks like the ads are pretty well obscured - the URL of the top one looks like `https://v.wpimg.pl/NzBrYi5...` for example.

Do you know if there are any ad blockers that block this, and perhaps what rule they're using if so?

**Reply by ghost** [reply]
Min browser ad blocking is getting better.This site I gave because it has a lot of ads.uBlock Origin can't handle it either ;-) Overall I am very satisfied with Min.

**Reply by ioc2e3** [reply]
> Do you have some examples of ads that Min doesn't block? It's possible there's something we can improve there.

For example , YouTube ...

YouTube ads still suddenly appear ...

.
